src/components/schedule-page/table/schedule/schedule-parts/base-cell/base-cell.models.ts

```diff
@@ -0,0 +1,42 @@
+import { VerboseDate } from "../../../../../../common-models/month-info.model";
+import { CellColorSet } from "../../../../../../helpers/colors/cell-color-set.model";
+import { BaseCellInputOptions } from "./base-cell-input.component";
+
+export type CellType = "cell" | "highlighted-cell";
+export const baseCellDataCy = (index: number, cellType: CellType): string => `${index}-${cellType}`;
+
+export enum CellManagementKeys {
+  Enter = "Enter",
+  Escape = "Escape",
+}
+
+export const PivotCellType = "Cell";
+
+export interface PivotCell {
+  type: string;
```

Comment:
a to nie powinien być typ PivotCellType, zamiast string. Albo trzeba będzie nazwy jakoś delikanie pozmieniać, bo nie rozumiem

Reply:
Racja
Tu był problem z nazewnictwem. 

Generalnie wybieranie zakresu działa tak, że biblioteka do drag and drop w momencie kiedy zaczynamy drag emituje jakiś event z danymi (`PivotCell`)
Potem każda komórka przez którą my przeciągamy ten `PivotCell` zmienia swój stan na zaznaczona/niezaznaczona.
Potrzeba było też zrobić tak, żeby jeśli my zaczynamy przeciąganie na jednej sekcji, to można było zaznaczać komórki tylko w obrębie tej sekcji.
Rozwiązuję to za pola `type` które jest ustawiane na klucz sekcji.
 Ale pomyślałem o tym, że mogło być tak, że klucz sekcji będzie użyty w jeszcze gdzieś do dragowania, więc postanowiłem, że będę dokładać do tego prefiks wskazujący na to, że event pochodzi z komórki która należy do sekcji.
Tylko, że tak jak mówisz, źle nazwałem ten klucz, więc teraz zmieniłem nazwę na `PivotCellTypePrefix`
Jak byś był ciekaw, to tak wygląda ta logika która opisywałem
```
const dragAnDropType = `${PivotCellTypePrefix}${sectionKey ?? ""}`;
  const [, drop] = useDrop({
    accept: dragAnDropType,
    collect: (monitor) => {
      if (monitor.isOver()) {
        if (!isBlocked) {
          onDrag?.(monitor.getItem() as PivotCell);
        }
      }
    },
    drop: () => {
      onDragEnd?.();
    },
  });
```